testsuite/cli/src/client_proxy.rs

```diff
@@ -679,60 +665,83 @@ impl ClientProxy {
         }
     }
 
-    /// Waits for the next transaction for a specific address and prints it
-    pub fn wait_for_transaction(
-        &mut self,
-        account: AccountAddress,
-        sequence_number: u64,
-    ) -> Result<()> {
-        let mut max_iterations = 5000;
-        println!(
-            "waiting for {} with sequence number {}",
-            account, sequence_number
-        );
-        loop {
-            stdout().flush().unwrap();
-
-            match self
-                .client
-                .get_txn_by_acc_seq(account, sequence_number - 1, true)
-            {
-                Ok(Some(txn_view)) => {
-                    println!();
-                    if txn_view.vm_status == VMStatusView::Executed {
-                        println!("transaction executed!");
-                        if txn_view.events.is_empty() {
-                            println!("no events emitted");
-                        }
-                        break Ok(());
-                    } else {
-                        break Err(format_err!(
-                            "transaction failed to execute; status: {:?}!",
-                            txn_view.vm_status
-                        ));
-                    }
-                }
-                Err(e) => {
-                    println!();
-                    println!("Response with error: {:?}", e);
-                }
-                _ => {
-                    print!(".");
+    /// Wait for transaction, this function is not safe for waiting for a specific transaction,
+    /// should use wait_for_signed_transaction instead.
+    /// TODO: rename to wait_for_account_seq or remove
+    pub fn wait_for_transaction(&self, address: AccountAddress, seq: u64) -> Result<()> {
+        let start = time::Instant::now();
+        while start.elapsed() < DEFAULT_WAIT_TIMEOUT {
+            let account_txn = self.client.get_txn_by_acc_seq(&address, seq, false)?;
+            if let Some(txn) = account_txn {
+                if txn.transaction.unwrap().sequence_number >= seq {
+                    return Ok(());
                 }
             }
-            max_iterations -= 1;
-            if max_iterations == 0 {
-                panic!("wait_for_transaction timeout");
+            std::thread::sleep(time::Duration::from_millis(10));
+        }
+        bail!(
+            "wait for account(address={}) transaction(seq={}) timeout",
+            address,
+            seq
+        )
+    }
+
+    /// Submit transaction and waits for the transaction executed
+    pub fn submit_and_wait(&mut self, txn: &SignedTransaction, is_blocking: bool) -> Result<()> {
+        self.client.submit_transaction(&txn)?;
+        if is_blocking {
+            self.wait_for_signed_transaction(txn)?;
+        } else {
+            self.update_account_seq(&txn.sender(), txn.sequence_number() + 1);
+        }
```

Comment:
this is testsuite, and it's also client code, so it's not the biggest deal. You can either ignore this line by adding `#[allow(clippy::integer_arithmetic)]` before it, or just replace the addition with `txn.sequence_number().checked_add(1).expect("seqnum can't reach u64::max");`